### PR TITLE
Ray Tune checkpointing fix, allow LR schedules for non-PCGrad opt, and more.

### PR DESCRIPTION
**Pull request by erwulff**
- Fix automatic saving and loading of model and optimizer checkpoints in Ray Tune runs
- Allow to use LR schedules when using optimizers other than PCGrad
- Add JUERCA sbatch script for multi-node Horovod training
- Use comet offline logging in Ray Tune experiments
- Etc.

### CODE REVIEW

**Review by erwulff — COMMENTED** [review]
Ready to merge from my side.

File: mlpf/pipeline.py [hunk 1]
```diff
@@ -194,7 +204,7 @@ def train(config, weights, ntrain, ntest, nepochs, recreate, prefix, plot_freq,
     model.fit(
         ds_train.repeat(),
         validation_data=ds_test.repeat(),
-        epochs=initial_epoch + config["setup"]["num_epochs"],
+        epochs=config["setup"]["num_epochs"],
```
Comment (jpata) [inline mlpf/pipeline.py:207]: why did you change this? I think it was correct.

Comment (erwulff) [reply mlpf/pipeline.py:207]: My thinking is the following. Let's say `config["setup"]["num_epochs"]` is `100` and we resume an interrupted training from epoch `20`. Then `initial_epoch` will be `20` and `initial_epoch + config["setup"]["num_epochs"]` will be 120, right? I think it's more intuitive that `config["setup"]["num_epochs"]` should be the total number of epochs to run before completing the training, rather than the additional number of epochs to run from the resumed point. This is a matter of taste I suppose. What do you think?

**Comment by jpata** [comment]
Thanks for this, looks good.

So I agree to change the meaning of epochs/nepochs from

"train X more epochs" to "train up to X epochs".
